test/end2end_test.go

```diff
@@ -7032,6 +7032,168 @@ func (s *httpServer) start(t *testing.T, lis net.Listener) {
 	}()
 }
 
+type testAuthInfo struct {
+	credentials.CommonAuthInfo
+	connection net.Conn
+}
+
+func (ta *testAuthInfo) AuthType() string {
+	return ""
+}
+
+// ConnWrapper wraps a net.Conn and pushes on a channel when closed.
+type ConnWrapper struct {
+	net.Conn
+	CloseCh *testutils.Channel
+}
+
+// Close closes the connection and sends a value on the close channel.
+func (cw *ConnWrapper) Close() error {
+	err := cw.Conn.Close()
+	cw.CloseCh.Replace(nil)
+	return err
+}
+
+type transportRestartCheckCreds struct {
+	mu          sync.Mutex
+	connections []ConnWrapper
+	credentials.TransportCredentials
```

Comment:
I see that you are implementing all methods of `credentials.TransportCredentials`. So, this embedding is not required. Else, if you can figure out that you can skip implementing some of these methods and the test still works fine, then you can embed the interface.